### ISSUE TITLE
Fix `declaration-block-no-redundant-longhand-properties` range for contiguous redundant longhand properties

### ISSUE THREAD

**Issue by pamelalozano16** [stylelint/stylelint#9272]
### What minimal example or steps are needed to reproduce the bug?

Currently, the `declaration-block-no-redundant-longhand-properties` rule only reports the position of the final redundant longhand property, rather than the full block.

When developers view this warning in their editor via LSP, only the last line is highlighted. This makes it difficult to quickly visualize the full scope of the redundant declarations, especially in larger blocks.

### What minimal configuration is needed to reproduce the bug?

```js
export default {
  extends: ['stylelint-config-standard'],
  rules: {
    "declaration-block-no-redundant-longhand-properties": true
  }
};
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdinKC2SATgOYCWmAtADYwBmALvCgCwAMADgB4DcKWu+ImUqFSxABaMUARg48+2XARLkK9COyYAmOb35KhqgEYR66vEwDMuhQF8QAGhC1SNAHJI8cRDE6f2NAB0YADOIY7gEJguxAggvuwQhPQosLRIAK5UKRiKvvQwmFAhTADaAOQh9ACeNFTk9BSQ0WIUVUhFRFDlALoO+oRZMCVo+jjoILBgVERI9KRRFEZUEGAA1hSYEBSEMFAZnZiNK5gSHVAU7IQaMMmkwxNM9IMw+rZYttxYEc0xAGJJAj0OJVWowepHYJRGKBPAAK3CTlg7HCiFy4xAoLqDUefBAMwKVQm-UUEyx4IaTWhrXanUIUFxEwJw3oEw+31sQA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

If the redundant longhand properties are contiguous (uninterrupted by other declarations), the reported warning range should span from the start of the first property to the end of the last property.

If the properties are separated by unrelated properties (e.g., a `color` declaration in the middle of `margin` declarations), it should safely fall back to highlighting only the final property to avoid falsely flagging unrelated code.

Expected editor highlighting:

```CSS
a {
    ^^^^^^^^^^^^^^^^^^^^
    margin-left: 40px; 
    margin-right: 10px; 
    margin-top: 20px; 
    margin-bottom: 30px;
    ^^^^^^^^^^^^^^^^^^^^
}
```

### What actually happened?

The rule attaches the warning strictly to the last node in the redundant sequence.
Example of current editor highlighting:

```CSS
a {
    margin-left: 40px; 
    margin-right: 10px; 
    margin-top: 20px; 
    margin-bottom: 30px; 
    ^^^^^^^^^^^^^^^^^ /* Only this line is flagged */
}
```

### Do you have a proposal to fix the bug?

Update the rule logic to check the relative indices of the collected longhand nodes within their parent block.

- If contiguous: Calculate the start offset of the first node and the end offset of the last node, reporting the full range on the parent block.

- If separated/interrupted: Retain the current behavior of reporting on the last trigger node.

*Note*: Ideally, whether the properties are contiguous or not shouldn't matter. If Stylelint's report function (and the underlying LSP) supported reporting multiple separated text ranges for a single diagnostic, we could just highlight the exact redundant lines no matter where they are.
